pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2541,7 +2541,25 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
   public ResultSet getFunctions(String catalog, String schemaPattern, String functionNamePattern)
       throws SQLException {
-    return getProcedures(catalog, schemaPattern, functionNamePattern);
+    String sql;
+    sql = "SELECT NULL AS FUNCTION_CAT, n.nspname AS FUNCTION_SCHEM, p.proname AS FUNCTION_NAME, "
+            + "d.description AS REMARKS, "
+            + DatabaseMetaData.procedureReturnsResult + " AS FUNCTION_TYPE, "
```

Comment:
Thanks @davecramer for this query tips, which give us more to play with. But this query only return user defined functions though, and the `getFunctions()` API javadoc says it should return both system + user function list.

Also, the `getFunctions()` API says one of the field: 

> FUNCTION_TYPE short => kind of function:
>    functionResultUnknown - Cannot determine if a return value or table will be returned
>    functionNoTable- Does not return a table
>    functionReturnsTable - Returns a table 

Would the "table" here means anything that will resulted in a "ResultSet" object return by the function? In that case, would result type of "refcursor" or "rowtype (table)" should be considered true or not?

For "refcursor" type we can check easily, but for this "rowtype (table)", the returned value is an actual table name. So this might complicate things.